WalletWasabi/Stores/IndexStore.cs

```diff
@@ -230,6 +230,43 @@ public async Task<FilterModel> RemoveLastFilterAsync(CancellationToken cancel)
 			return filter;
 		}
 
+		public async Task<IEnumerable<FilterModel>> RemoveAllImmmatureFiltersAsync(CancellationToken cancel, bool deleteAndCrashIfMature = false)
+		{
+			var removed = new List<FilterModel>();
+			using (await IndexLock.LockAsync(cancel))
+			{
+				if (ImmatureFilters.Any())
+				{
+					Logger.LogWarning<IndexStore>($"Filters got corrupted. Reorging {ImmatureFilters.Count} immature filters in an attempt to fix them.");
+				}
+				else
+				{
+					Logger.LogCritical<IndexStore>($"Filters got corrupted and have no more immature filters.");
+
+					if (deleteAndCrashIfMature)
+					{
+						Logger.LogCritical<IndexStore>($"Deleting all filters and crashing the software...");
+
+						using (await MatureIndexFileManager.Mutex.LockAsync(cancel))
+						using (await ImmatureIndexFileManager.Mutex.LockAsync(cancel))
+						{
+							ImmatureIndexFileManager.DeleteMe();
+							MatureIndexFileManager.DeleteMe();
+						}
+
+						Environment.Exit(2);
+					}
+				}
+			}
+
+			while (ImmatureFilters.Any())
+			{
+				removed.Add(await RemoveLastFilterAsync(cancel));
```

Comment:
I didn't test it but I think this has to be very hard to test because the `RemoveLastFilterAsync` does a lot of things and try to save to file.